parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -731,9 +731,19 @@ def worker(
         os.sched_setaffinity(0, my_cores)  # type: ignore[attr-defined, unused-ignore]
         logger.info("Set worker CPU affinity to {}".format(my_cores))
 
+    # If CUDA devices, find total number of devices to allow for MPS
+    # See: https://developer.nvidia.com/system-management-interface
+    try:
+        mod_factor = int(os.popen("nvidia-smi -L | wc -l").read())
+    except:
+        mod_factor = None
+        
     # If desired, pin to accelerator
     if accelerator is not None:
-        os.environ["CUDA_VISIBLE_DEVICES"] = accelerator
+        try:
+            os.environ["CUDA_VISIBLE_DEVICES"] = str(int(accelerator)%mod_factor) # multiple workers will share a GPU
+        except:
```

Comment:
The only time this should fail is if `mod_factor` is None. So, what about using `if mod_factor is not None ... else` instead of `try ... catch`?

Reply:
The one other time it would fail is if `accelerator` can't be cast as an `int`.  I'm focused on Polaris and Perlmutter for this development.  I'm not sure if there is a situation where `CUDA_VISIBLE_DEVICES` would be set to a non-integer, at least, I'm not aware of one.  So I think any situation that accelerator can't be cast as an `int`, `mod_factor` should also be `None`, so I think you are right, but if anyone knows better, please correct me.

Reply:
So if someone wanted to set a worker to use 2 GPUS, accelerator should be `"0,1"` or `"2,3"`.

Reply:
In that case, we want this to fail and fall back to using whatever string `accelerator` was set to.

Reply:
Nice! I forgot that it could be a string.